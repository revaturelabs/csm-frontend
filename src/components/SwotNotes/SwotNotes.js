import React from "react";
import { Form, TextArea, Header } from "semantic-ui-react";
import { useDispatch } from "react-redux";

const SwotNotes = (props) => {
  const dispatch = useDispatch();

  return (
    <>
      <Form>
        <Header as='h3'
        style={{padding: '10px'}}>Notes</Header>
        <TextArea
          placeholder="Notes..."
          onChange={(e) =>
            dispatch({ type: "updateNotes", notes: e.target.value })
          }
        />
      </Form>
    </>
  );
};

export default SwotNotes;
