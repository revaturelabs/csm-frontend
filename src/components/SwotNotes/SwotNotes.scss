@import '../../styles/main.scss';

textarea.swot-overall-notes {
  height: 5rem !important;
  transition: height .2 ease-in-out !important;

  &:focus, &:active {
    height: 15rem !important;
    transition: height .2 ease-in-out !important;
  }
}