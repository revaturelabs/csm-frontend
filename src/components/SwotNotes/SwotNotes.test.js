// Component.test.js
import React from "react";
import ReactDOM from "react-dom";
import Enzyme, { shallow, mount, render, fireEvent } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { useSelector, useDispatch, useShallowEqualSelector } from 'react-redux';
import configureStore from 'redux-mock-store';
import reducer from '../../reducers';
import SwotNotes from './SwotNotes';

// It can receive two more parameters, the second one is to specify a factory instead of the jest's automocking feature
jest.mock('react-redux');

describe("SwotNotes unit test", () => {

	it('Rendering component without props.', () => {
		const component = shallow(<SwotNotes/>);
		expect(component).toMatchSnapshot();
	});

	it('Rendering component with props that match the original component.', () => {
		// Set your own props here...
		const props = "props";
		const component = render(<SwotNotes props={props} />);
		expect(component).toMatchSnapshot();
	});

	it('Mounting, testing user input, and dismounting the component.', () => {
		// Set your own functions and props here...
		props = "props";
		find = jest.fn();
		const component = mount(<Search id="find" props={props} find={find} />);
		component.find('#find').value;
		component.simulate('keypress', { keyCode: 'A' });
		expect(component).toMatchSnapshot();
		component.unmount();
	});

});
