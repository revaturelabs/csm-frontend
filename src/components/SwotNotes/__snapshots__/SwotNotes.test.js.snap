// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwotNotes unit test Rendering component without props. 1`] = `
<Fragment>
  <Form
    as="form"
  >
    <Header
      as="h3"
      style={
        Object {
          "padding": "10px",
        }
      }
    >
      Notes
    </Header>
    <TextArea
      as="textarea"
      onChange={[Function]}
      placeholder="Notes..."
      rows={3}
    />
  </Form>
</Fragment>
`;
